### PR TITLE
DASD:8323 - Added failure-anomalies ARM template and Get-AppInsightsResources.ps1

### PR DESCRIPTION
**Pull request by jack-education**

### CODE REVIEW

File: tools/Get-AppInsightsResources.ps1 [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    $AppInsightsResourceObject | Add-Member -NotePropertyMembers @{
+        appInsightsName             = $AppInsightsResource.Name
+        appInsightsResourceGroup    = $AppInsightsResource.ResourceGroupName
+    }
+    $AppInsightsResourcesArray += $AppInsightsResourceObject
+}
+
+$AppInsightsResourcesArrayString = $AppInsightsResourcesArray | ConvertTo-Json -Compress
+
+Write-Output ("##vso[task.setvariable variable=APP_INSIGHTS_RESOURCES_ARRAY_STRING]$($AppInsightsResourcesArrayString)")
```
Comment (NickGraham101) [inline tools/Get-AppInsightsResources.ps1:21]: Why are the outer most set of braces needed?

File: azure/failure-anomalies.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+          },
+          "severity": {
+            "value": "Sev0"
+          }
+        }
+      },
+      "copy": {
+        "name": "failureAnomaliesLoop",
+        "count": "[length(parameters('appInsightsResourcesArray'))]",
+        "mode": "Parallel"
```
Comment (NickGraham101) [inline azure/failure-anomalies.json:45]: Is there a risk of conflicting deployments where a resource group contains multiple app insights?

File: tools/Get-AppInsightsResources.ps1 (hunk 1 above)
Comment (jack-education) [reply tools/Get-AppInsightsResources.ps1]: Good spot, I've removed them

File: azure/failure-anomalies.json (hunk 2 above)
Comment (jack-education) [reply azure/failure-anomalies.json:45]: The deployment name is unique, updated to failure-anomalies-<appInsightsName> for clarity, and hasn't errored with my two app insights resources in my tenant so don't think there should be conflicts

Comment (jack-education) [reply azure/failure-anomalies.json:45]: Successfully deployed to a dev environment

**Review by NickGraham101 — APPROVED** [review]

**Pull request merged**